code/game/machinery/nebula_shielding.dm

```diff
@@ -0,0 +1,130 @@
+///Machinery that block nebula's. This type is for convenience, you can set nebula shielding on other objects aswell using add_to_nebula_shielding()
+/obj/machinery/nebula_shielding
+	density = TRUE
+
+	icon = 'icons/obj/machines/nebula_shielding.dmi'
+	pixel_x = -16
+
+	layer = FLY_LAYER
+	plane = ABOVE_GAME_PLANE
+
+	///Strength of the shield we apply
+	var/shielding_strength
+	///The type of nebula that we shield against
+	var/nebula_type
+	///How much power we use every time we block the nebula's effects
+	var/power_use_per_block = BASE_MACHINE_ACTIVE_CONSUMPTION * 2
+	///State we use when actively blocking a nebula
+	var/active_icon_state
+
+/obj/machinery/nebula_shielding/Initialize(mapload)
+	. = ..()
+
+	add_to_nebula_shielding(src, nebula_type, PROC_REF(get_nebula_shielding))
+
+///Nebula is asking us how strong we are. Return our shield strength is all is well
+/obj/machinery/nebula_shielding/proc/get_nebula_shielding()
+	SIGNAL_HANDLER
+	if(panel_open)
+		return
+	if(!powered())
+		icon_state = initial(icon_state)
+		return
+
+	use_power_from_net(power_use_per_block)
+	generate_reward()
+	icon_state = active_icon_state
+	return shielding_strength
```

Comment:
You cleared Melbert's comment but I will repeat--**signals should not return anything other than bitfields**.

Otherwise you cannot have more than one thing register to them and change values.

Reply:
It's a callback, not a signal. I am completely clueless as to why I didnt remove the SIGNAL_HANDLER here, I swore there was none